src/category_theory/limits/shapes/pullbacks.lean

```diff
@@ -0,0 +1,244 @@
+-- Copyright (c) 2018 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+
+import category_theory.eq_to_hom
+import category_theory.limits.cones
+import category_theory.sparse
+
+open category_theory
+
+namespace tactic
+meta def case_bash : tactic unit :=
+do l ← local_context,
+   r ← successes (l.reverse.map (λ h, cases h >> skip)),
+   when (r.empty) failed
+end tactic
+
+namespace category_theory.limits
+
+universes v u
+
+local attribute [tidy] tactic.case_bash
+
+@[derive decidable_eq] inductive walking_cospan : Type v
+| left | right | one
+@[derive decidable_eq] inductive walking_span : Type v
+| zero | left | right
+
+open walking_cospan
+open walking_span
+
+inductive walking_cospan_hom : walking_cospan → walking_cospan → Type v
+| inl : walking_cospan_hom left one
+| inr : walking_cospan_hom right one
+| id : Π X : walking_cospan.{v}, walking_cospan_hom X X
+inductive walking_span_hom : walking_span → walking_span → Type v
+| fst : walking_span_hom zero left
+| snd : walking_span_hom zero right
+| id : Π X : walking_span.{v}, walking_span_hom X X
+
+open walking_cospan_hom
+open walking_span_hom
+
+def walking_cospan_comp :
+  Π (X Y Z : walking_cospan) (f : walking_cospan_hom X Y) (g : walking_cospan_hom Y Z),
+    walking_cospan_hom X Z
+| _ _ _ (id _) h := h
+| _ _ _ inl    (id one) := inl
+| _ _ _ inr    (id one) := inr
+.
+
+def walking_span_comp :
+  Π (X Y Z : walking_span) (f : walking_span_hom X Y) (g : walking_span_hom Y Z),
+    walking_span_hom X Z
+  | _ _ _ (id _) h := h
+  | _ _ _ fst    (id left) := fst
+  | _ _ _ snd    (id right) := snd
+.
+
+instance walking_cospan_category_struct : category_struct walking_cospan :=
+{ hom  := walking_cospan_hom,
+  id   := walking_cospan_hom.id,
+  comp := walking_cospan_comp, }
+
+instance walking_cospan_sparse (X Y : walking_cospan) : subsingleton (X ⟶ Y) :=
```

Comment:
I'd rather call this `walking_cospan.subsingleton`, I think. Also, can't `tidy` + `case_bash` take care of the proof?

Reply:
I've moved this into a namespace, and removed the name.

I've also removed the proof, thanks.